### DISCUSSION TITLE
关于请求和通知无法被正常处理

### DISCUSSION THREAD

**Discussion by HeyYubadboy** [nonebot/discussions#23] [问答] [answered]
在我根据Nonebot的文档写自动同意和群成员欢迎的时候，我发现Nonebot在Debug中Event的user_id是机器人的id，而[文档中](https://github.com/botuniverse/onebot#/Post?id=%E5%8A%A0%E5%A5%BD%E5%8F%8B%E8%AF%B7%E6%B1%82)规定user_id是请求加好友的人的id，并且我发现@on_notice并没有在群成员增加时被调用，并且可以说nonebot根本没有检测到
使用的工具是：先驱+nonebot

**Comment by HeyYubadboy** [comment]
文档链接是：
https://github.com/botuniverse/onebot#/Post?id=%E5%8A%A0%E5%A5%BD%E5%8F%8B%E8%AF%B7%E6%B1%82

https://docs.nonebot.dev/guide/notice-and-request.html#%E6%AC%A2%E8%BF%8E%E6%96%B0%E6%88%90%E5%91%98

**Comment by yanyongyu** [comment, marked as answer]
nonebot日志中的event为事件上报原始内容，如果user_id存在错误请询问先驱onebot

> on_notice会在群成员增加时调用，并且可以说它根本没有检测到

请提供相关日志

**Reply by HeyYubadboy** [reply]
[2021-08-11 15:54:13,389 nonebot] INFO: Request: <Event, {'comment': '', 'flag': 2581487884, 'post_type': 'request', 'request_type': 'friend', 'self_id': 1806267337, 'time': 0, 'user_id': 1806267337}>
[2021-08-11 15:54:13,389 nonebot] DEBUG: Emitting event: request.friend

**Reply by yanyongyu** [reply]
这是request事件不是notice事件

**Comment by mnixry** [comment]
> 我发现Nonebot在Debug中Event的user_id是机器人的id，而文档中规定user_id是请求加好友的人的id

NoneBot会**原样序列化**传入的event, 如果event字段与规范不符, 请前往OneBot实现部分(如XQHTTP?)提交issue

> 并且我发现@on_notice并没有在群成员增加时被调用，并且可以说nonebot根本没有检测到

请求加好友的事件类型属于`request`而不是`notice`<sup>[参考文档](https://github.com/botuniverse/onebot/blob/master/v11/specs/event/request.md#%E5%8A%A0%E5%A5%BD%E5%8F%8B%E8%AF%B7%E6%B1%82)</sup>
